DownloadProject.cmake

```diff
@@ -94,29 +95,35 @@ function(download_project)
         unset(OUTPUT_QUIET)
         message(STATUS "Downloading/updating ${DL_ARGS_PROJ}")
     endif()
-
+    
+    if(NOT DEFAULT_PARENT_DIR)
+       set(DEFAULT_PARENT_DIR  "${CMAKE_BINARY_DIR}/thirdparty" )
+    endif()
```

Comment:
In this test, DEFAULT_PARENT_DIR will always be empty (unless you were unlucky and something at parent scope set it). You probably meant to use DL_ARGS_DEFAULT_PARENT_DIR, since that's what will be populated by `cmake_parse_arguments()`.